src/Orleans/Messaging/GatewayManager.cs

```diff
@@ -253,7 +253,8 @@ private void UpdateLiveGatewaysSnapshot(IEnumerable<Uri> currentKnownGateways, T
 
         public void Dispose()
         {
-            gatewayRefreshTimer?.Dispose();
+            if (gatewayRefreshTimer != null)
+                gatewayRefreshTimer.Dispose();
```

Comment:
yup, that was what I suggested in an email to him. I'd be fine with either (copy value to variable, or explicit supression of warning), but prefer not to risk race conditions